xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift

```diff
@@ -28,12 +29,40 @@ class CourseDecisionViewController: UIViewController {
         NSNotificationCenter.defaultCenter().removeObserver(self);
     }
 
+    @IBAction func enroll(sender: UIBarButtonItem) {
+        if UserProfileHelper.isLoggedIn() {
+            UserProfileHelper.createEnrollement(course.id).onSuccess {
+                CourseHelper.refreshCourses().onSuccess(callback: { _ in
```

Comment:
`onSuccess` should be called on the main thread anyway. Did you verify `dispatch_async` is necessary here?